aws_s3_policies/aws_s3_bucket_secure_access.py

```diff
@@ -3,17 +3,51 @@
 from panther_base_helpers import deep_get
 from policyuniverse.policy import Policy
 
+# According to AWS there should exist an explicit deny policy that contains
+# "aws:SecureTransport": "false" for this check to be compliant
+# https://aws.amazon.com/premiumsupport/knowledge-center/s3-bucket-policy-for-config-rule/
+# This policy returns a LOW severity alert if the bucket policy is using an implicit deny
+# of secure transport or a HIGH severity if secure transport is not enforced.
+
+EXPLICIT_DENY = False
+IMPLICIT_DENY = True
+
 
 def policy(resource):
     if resource["Policy"] is None:
         return False
 
+    global EXPLICIT_DENY, IMPLICIT_DENY  # pylint: disable=global-statement
+
     iam_policy = Policy(json.loads(resource["Policy"]))
 
     for statement in iam_policy.statements:
-        if statement.effect != "Allow":
-            continue
-        if not deep_get(statement.statement, "Condition", "Bool", "aws:SecureTransport"):
-            return False
+        if (
+            statement.effect == "Deny"
+            and deep_get(statement.statement, "Condition", "Bool", "aws:SecureTransport") != "true"
+        ):
+            EXPLICIT_DENY = True
+            break
+
+        if (
+            statement.effect == "Allow"
+            and deep_get(statement.statement, "Condition", "Bool", "aws:SecureTransport") != "true"
+        ):
+            IMPLICIT_DENY = False
+
+    if IMPLICIT_DENY and not EXPLICIT_DENY:
+        return False
+
+    return EXPLICIT_DENY
```

Comment:
I wonder if we can simplify this by only returning `EXPLICIT_DENY`? if there is not explicit deny, we should trigger an alert. And then in the severity function, we use whether or not IMPLICIT_DENY is set to change the severity / title? 

```suggestion
    return EXPLICIT_DENY 
```

Reply:
Removing this breaks the test case `Bucket Uses Implicit Allow AND Explicit Deny` but I'm starting to think thats not a good test.  In the off chance that a bucket has both things the Deny wins, so it doesn't really matter if they have both.  

Thoughts?

Reply:
Good call, this was also masking a bad test I created that tried to pass in 2 policies 🤦🏼 . Updating and removing that test